Exiled.Events/Patches/Events/Player/ChangingRole.cs

```diff
@@ -70,12 +70,17 @@ private static bool Prefix(CharacterClassManager __instance, ref RoleType classi
                     classid = escapingEventArgs.NewRole;
                 }
 
+                var oldRole = API.Features.Player.Get(ply).Role;
```

Comment:
Get the player once instead, don't call `Player::Get(GameObject)` every time you need a value of a property.